kafka-client/src/main/java/dev/responsive/kafka/internal/stores/CommitBuffer.java

```diff
@@ -370,7 +371,9 @@ public KeyValueIterator<K, Result<K>> all(
   ) {
     return Iterators.kv(
         Iterators.filter(
-            buffer.getReader().entrySet().iterator(),
+            // make a full copy to avoid concurrent mod exceptions
+            // we can worry about performance for all() some other time
+            new TreeMap<>(buffer.getReader()).entrySet().iterator(),
```

Comment:
probably better to use `ImmutableMap.copyOf`

Reply:
I reverted this from this patch since I no longer need it and I'm not 100% sure it was necessary